helpers/getNextConfig.js

```diff
@@ -1,28 +1,29 @@
+'use strict'
+
 const { resolve } = require('path')
 
-let nextConfig
+const moize = require('moize')
 
 // Load next.config.js
 const getNextConfig = async function (failBuild = defaultFailBuild) {
-  // Memoizes `nextConfig`
-  if (nextConfig !== undefined) {
-    return nextConfig
+  try {
+    return await moizedGetNextConfig()
+  } catch (error) {
+    return failBuild('Error loading your next.config.js.', { error })
   }
+}
 
+const getNextConfigValue = function () {
   // We cannot load `next` at the top-level because we validate whether the
   // site is using `next` inside `onPreBuild`.
   const { PHASE_PRODUCTION_BUILD } = require('next/constants')
   const loadConfig = require('next/dist/next-server/server/config').default
 
-  try {
-    nextConfig = await loadConfig(PHASE_PRODUCTION_BUILD, resolve('.'))
-  } catch (error) {
-    return failBuild('Error loading your next.config.js.', { error })
-  }
-
-  return nextConfig
+  return loadConfig(PHASE_PRODUCTION_BUILD, resolve('.'))
 }
 
+const moizedGetNextConfig = moize(getNextConfigValue, { maxSize: 1e3, isPromise: true })
```

Comment:
`maxSize` is required by `moize@6`. `1` would work in our case since the function has no arguments, but I used `1e3` so it covers future usage where we might add more arguments to this function.